src/Kis/Kis.hs

```diff
@@ -2,39 +2,69 @@
 {-# LANGUAGE OverloadedStrings          #-}
 {-# LANGUAGE RankNTypes                 #-}
 {-# LANGUAGE StandaloneDeriving         #-}
+{-# LANGUAGE TypeSynonymInstances       #-}
+{-# LANGUAGE FlexibleInstances          #-}
 
 module Kis.Kis
     ( withKis
-    , KisAction(..)
+    , KisRequest(..)
     , KisClient
     , KisConfig(..)
     , Kis(..)
     , KisException(..)
+    , KisT
+    , createBed
+    , createPatient
+    , getPatients
+    , getPatient
+    , placePatient
+    , runKis
     )
 where
 
 import Control.Monad.Logger
+import Control.Monad.Operational
 import Control.Monad.RWS hiding (asks)
 import Control.Monad.Reader
 import Database.Persist
-import Data.Text
+import Data.Text hiding (singleton)
 import GHC.Exception
 
 import Kis.Model
 import Kis.Time
 
-data KisAction a where
-    CreateBed :: String -> KisAction BedId
-    CreatePatient :: Patient -> KisAction PatientId
-    GetPatient :: PatientId -> KisAction (Maybe Patient)
-    GetPatients :: KisAction [Entity Patient]
-    PlacePatient :: PatientId -> BedId -> KisAction (Maybe PatientBedId)
+data KisRequest a where
+    CreateBed :: String -> KisRequest BedId
+    CreatePatient :: Patient -> KisRequest PatientId
+    GetPatient :: PatientId -> KisRequest (Maybe Patient)
+    GetPatients :: KisRequest [Entity Patient]
+    PlacePatient :: PatientId -> BedId -> KisRequest (Maybe PatientBedId)
 
-deriving instance Show (KisAction a)
+type KisT m = ProgramT KisRequest m
+
+runKis :: KisT IO a -> IO a
+runKis = undefined
+
+class Monad m => MonadKis m where
+    createBed :: String -> m BedId
+    createPatient :: Patient -> m PatientId
+    getPatients :: m [Entity Patient]
+    getPatient :: PatientId -> m (Maybe Patient)
+    placePatient :: PatientId -> BedId -> m (Maybe PatientBedId)
+
+instance Monad m => MonadKis (KisT m) where
+    createPatient = singleton . CreatePatient
+    createBed = singleton . CreateBed
+    getPatients = singleton GetPatients
+    getPatient = singleton . GetPatient
+    placePatient patId bedId = singleton (PlacePatient patId bedId)
+
+
+deriving instance Show (KisRequest a)
 
```

Comment:
Irgendwie habe ich das Gefühl, hier kommt viel Komplexität dazu, ohne dass es viel Gewinn bringt. Der Typ KisT m ist jetzt vielleicht einfacher, dafür haben wir irgendwie das Interface zwei mal definiert, einmal im GADT und einmal in der Typklasse. Vielleicht muss ich erst verstehen, warum die zwei sich unterscheiden. 

Cool finde ich, dass man damit dann kompliziertere Operationen als Kombination der grundlegenden Operationen aus KisRequest bauen kann und die generische zu Verfügung stellen kann. Das wird später sicher wichtig werden. Andererseits konnten wir das davor ja auch, als Funktionen in (KisClient m a), die sind ja auch erstmal generisch.